platform/api/oci/signer_test.go

```diff
@@ -0,0 +1,131 @@
+		postCall(t, signer)
+	})
+}
+
+func getCall(t *testing.T, signer *Signer) {
+	uri := "https://iaas.us-phoenix-1.oraclecloud.com/20160918/instances?availabilityDomain=Pjwf%3A%20PHX-AD-1&compartmentId=ocid1.compartment.oc1..aaaaaaaam3we6vgnherjq5q2idnccdflvjsnog7mlr6rtdb25gilchfeyjxa&displayName=TeamXInstances&volumeId=ocid1.volume.oc1.phx.abyhqljrgvttnlx73nmrwfaux7kcvzfs3s66izvxf2h4lgvyndsdsnoiwr5q"
+
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		t.Fatalf("creating request: %v", err)
+	}
+
+	req.Header.Set("date", "Thu, 05 Jan 2014 21:31:40 GMT")
+
+	err = signer.SignRequest(req)
+	if err != nil {
+		t.Fatalf("signing request: %v", err)
+	}
+
+	auth := req.Header.Get("Authorization")
+
+	RegexpContains(t, auth, "keyId=\"ocid1.tenancy.oc1..aaaaaaaaba3pv6wkcr4jqae5f15p2b2m2yt2j6rx32uzr4h25vqstifsfdsq/ocid1.user.oc1..aaaaaaaat5nvwcna5j6aqzjcaty5eqbb6qt2jvpkanghtgdaqedqw3rynjq/20:3b:97:13:55:1c:5b:0d:d3:37:d8:50:4e:c5:3a:34\"", "keyId")
+	RegexpContains(t, auth, "algorithm=\"rsa-sha256\"", "algorithm")
+	RegexpContains(t, auth, "signature=\"GBas7grhyrhSKHP6AVIj/h5/Vp8bd/peM79H9Wv8kjoaCivujVXlpbKLjMPeDUhxkFIWtTtLBj3sUzaFj34XE6YZAHc9r2DmE4pMwOAy/kiITcZxa1oHPOeRheC0jP2dqbTll8fmTZVwKZOKHYPtrLJIJQHJjNvxFWeHQjMaR7M=\"", "signature")
+	RegexpContains(t, auth, "version=\"1\"", "version")
+	RegexpContains(t, auth, "headers=\"date \\(request-target\\) host\"", "headers")
+}
+
+func postCall(t *testing.T, signer *Signer) {
+	uri := "https://iaas.us-phoenix-1.oraclecloud.com/20160918/volumeAttachments"
+
+	req, err := http.NewRequest("POST", uri, nil)
+	req.Body = ioutil.NopCloser(bytes.NewReader([]byte(`{
+    "compartmentId": "ocid1.compartment.oc1..aaaaaaaam3we6vgnherjq5q2idnccdflvjsnog7mlr6rtdb25gilchfeyjxa",
+    "instanceId": "ocid1.instance.oc1.phx.abuw4ljrlsfiqw6vzzxb43vyypt4pkodawglp3wqxjqofakrwvou52gb6s5a",
+    "volumeId": "ocid1.volume.oc1.phx.abyhqljrgvttnlx73nmrwfaux7kcvzfs3s66izvxf2h4lgvyndsdsnoiwr5q"
+}`)))
+
+	req.Header.Set("date", "Thu, 05 Jan 2014 21:31:40 GMT")
+
+	err = signer.SignRequest(req)
+	if err != nil {
+		t.Fatalf("signing request: %v", err)
+	}
+
+	auth := req.Header.Get("Authorization")
+
+	RegexpContains(t, auth, "keyId=\"ocid1.tenancy.oc1..aaaaaaaaba3pv6wkcr4jqae5f15p2b2m2yt2j6rx32uzr4h25vqstifsfdsq/ocid1.user.oc1..aaaaaaaat5nvwcna5j6aqzjcaty5eqbb6qt2jvpkanghtgdaqedqw3rynjq/20:3b:97:13:55:1c:5b:0d:d3:37:d8:50:4e:c5:3a:34\"", "keyId")
+	RegexpContains(t, auth, "algorithm=\"rsa-sha256\"", "algorithm")
+	RegexpContains(t, auth, "signature=\"Mje8vIDPlwIHmD/cTDwRxE7HaAvBg16JnVcsuqaNRim23fFPgQfLoOOxae6WqKb1uPjYEl0qIdazWaBy/Ml8DRhqlocMwoSXv0fbukP8J5N80LCmzT/FFBvIvTB91XuXI3hYfP9Zt1l7S6ieVadHUfqBedWH0itrtPJBgKmrWso=\"", "signature")
+	RegexpContains(t, auth, "version=\"1\"", "version")
+	RegexpContains(t, auth, "headers=\"date \\(request-target\\) host content-length content-type x-content-sha256\"", "headers")
+
+	if req.Header.Get("x-content-sha256") != "V9Z20UJTvkvpJ50flBzKE32+6m2zJjweHpDMX/U4Uy0=" {
```

Comment:
Maybe use a variable to avoid repeating the hash string, or drop it from the Fatalf since it's hardcoded anyway.